http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -107,6 +113,7 @@
     private final List<HttpRequestFilter> requestFilters;
     private final Exception creationLocation = new Exception();
     private final String name;
+    private final ClientSecurityConfig securityConfig;
```

Comment:
This doesn't need to hold on to the entire `ClientSecurityConfig` object. It just needs a flag to indicate whether security is enabled (for the check in `buildJettyRequest`)